src/server/generic_family.cc

```diff
@@ -101,14 +101,57 @@ ::io::Result<size_t> InMemSource::ReadSome(const iovec* v, uint32_t len) {
   return read_total;
 }
 
+class RestoreArgs {
+ private:
+  static constexpr time_t NO_EXPIRATION = 0;
+
+  time_t expiration_ = NO_EXPIRATION;
+  bool abs_time_ = false;
+  bool replace_ = false;  // if true, over-ride existing key
+  bool sticky_ = false;
+
+ public:
+  RestoreArgs() = default;
+
+  RestoreArgs(time_t expiration, bool abs_time, bool replace)
+      : expiration_(expiration), abs_time_(abs_time), replace_(replace) {
+  }
+
+  constexpr bool Replace() const {
+    return replace_;
+  }
+
+  constexpr bool Sticky() const {
+    return sticky_;
+  }
```

Comment:
This is the only change in this class (along with adding `sticky_` in line 111), otherwise I just moved this class to be above `RdbRestoreValue` so that the latter can use it